text/0000-ember-data-singleton-record-data.md

```diff
@@ -0,0 +1,450 @@
+receive a deprecation notice the first time an instance of
+a previously unseen `RecordData` class is encountered.
+
+Note that to comply with `SemVer`, any given release of `ember-data` will
+only have support for a maximum of two versions of `RecordData`, with
+deprecated versions being dropped following LTS releases.
+
+The APIs proposed by this RFC constitute version `"2"`
+and will be used by `ember-data` when the following is
+`true`.
+
+```js
+recordData.version === "2";
+```
+
+It is possible that once the `RecordDataManager` is in place and this upgrade to `identifiers` has
+occurred that we never need to increment this `version` again. If so, a future RFC may choose to
+deprecate the `version` property altogether once "version 1" is no longer supported.
+
+### RecordDataManager
+
+The `RecordDataManager` would implement the `RecordData` interface including all
+deprecated methods and deprecated method signatures. The manager is responsible for
+
+- Managing deprecations
+- Version translation
+- Internal Book Keeping
+
+Beginning with this RFC, `StoreWrapper.recordDataFor` will return the instance of
+a `RecordDataManager` for a given `RecordData` instead of the `RecordData` itself.
+
+This encapsulates the `RecordData` feature ensuring that communication is via public API
+and provides us the ability to manage the three concerns listed above and described below.
+
+**Managing Deprecations**
+
+When a `RecordData` method or method signature is deprecated, the `RecordDataManager` would be
+responsible for detecting calls to deprecated methods or using deprecated method signatures
+and printing the appropriate deprecation messages.
+
+**Version Translation**
+
+When a `RecordData` method or method signature is deprecated the `target` may not have
+implemented the deprecated method or signature. The `RecordDataManager` would use the
+combination of calling context, presence of the method on the `target` and the declared
+support `version` of the `target` to convert calls to deprecated methods or signatures
+into their supported modern equivalent.
+
+**Internal Book Keeping**
+
+In some situations `ember-data` will need to internally track of specific kinds of requests
+or operations sent to a `RecordData` instance. The `Manager` is our "intercept" point at
+which such tracking would be handled. Notably `RequestManager` will never fail to call
+the appropriate method with the appropriate arguments on the `target`, it will merely
+make notes about what has been called. E.g. `RequestManager` allows for `ember-data` to
+introduce `stateless side-effects` for some method calls, but it will not alter the original call
+or response beyond translating calls for deprecations. `Stateless` means that these `side-effects`
+will not alter any `application state` and have largely invisible effects on the data layer.
+From the perspective of a `RecordData` author these methods should be considered `stateless`.
```

Comment:
Is this paragraph intended to be about `RecordDataManager` and not `RequestManager`?

Reply:
🤦‍♂